Step03/app/src/main/java/com/kwoolytech/step03/HansWeatherDataModel.java

```diff
@@ -0,0 +1,84 @@
+package com.kwoolytech.step03;
+
+import android.app.Activity;
+import android.content.Context;
+import android.util.Log;
+import android.widget.ListView;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class HansWeatherDataModel {
+    private Context            mainContext;
+    private KwoolyHttp         httpClient;
+    private KwoolyHttpCallback httpClientCallback;
+    private JSONObject         weatherData;
+
+    HansWeatherDataModel(Context context) {
+        mainContext = context;
+        httpClient  = new KwoolyHttp(mainContext);
+        weatherData = new JSONObject();
+
+        httpClientCallback = new KwoolyHttpCallback() {
+            @Override
+            public void getJsonDataCallback(String result) {
+                try {
+                    weatherData = new JSONObject(result);
+                    ((ListView)((Activity) mainContext).findViewById(R.id.listViewWeather)).invalidateViews();
+                } catch (JSONException e) {
```

Comment:
I think invalidateViews() here just redraw the view itself instead of updating it with new data. So use below API to refresh UI with new data. https://developer.android.com/reference/android/widget/BaseAdapter.html#notifyDataSetChanged()